### PR TITLE
[WFLY-10365] Generate the component-matrix

### PR DESCRIPTION
**Pull request by kabir**
https://issues.jboss.org/browse/WFLY-10365

Output from build:
```
$mvn clean install -pl component-matrix-builder/
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building WildFly: Component matrix builder 13.0.0.Alpha1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ wildfly-component-matrix-builder ---
[INFO] Deleting /Users/kabir/sourcecontrol/wildfly/git/wildfly/component-matrix-builder/target
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (ban-bad-dependencies) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-java-version) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven-version) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- buildnumber-maven-plugin:1.4:create (get-scm-revision) @ wildfly-component-matrix-builder ---
[INFO] Executing: /bin/sh -c cd '/Users/kabir/sourcecontrol/wildfly/git/wildfly/component-matrix-builder' && 'git' 'rev-parse' '--verify' 'HEAD'
[INFO] Working directory: /Users/kabir/sourcecontrol/wildfly/git/wildfly/component-matrix-builder
[INFO] Storing buildNumber: 14932f15e391c4d41e42e8ec0cfa7c422fb6adfc at timestamp: 1526313503587
[INFO] Storing buildScmBranch: WFLY-10365-generate-bom
[INFO] 
[INFO] --- wildfly-component-matrix-plugin:1.0.1.Final:build-bom (build-bom) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.0.0:checkstyle (check-style) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- maven-source-plugin:3.0.1:jar-no-fork (attach-sources) @ wildfly-component-matrix-builder ---
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ wildfly-component-matrix-builder ---
[INFO] Installing /Users/kabir/sourcecontrol/wildfly/git/wildfly/component-matrix-builder/pom.xml to /Users/kabir/.m2/repository/org/wildfly/wildfly-component-matrix-builder/13.0.0.Alpha1-SNAPSHOT/wildfly-component-matrix-builder-13.0.0.Alpha1-SNAPSHOT.pom
[INFO] Installing /Users/kabir/sourcecontrol/wildfly/git/wildfly/component-matrix-builder/target/bom-pom.xml to /Users/kabir/.m2/repository/org/wildfly/wildfly-component-matrix/13.0.0.Alpha1-SNAPSHOT/wildfly-component-matrix-13.0.0.Alpha1-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.808 s
[INFO] Finished at: 2018-05-14T16:58:24+01:00
[INFO] Final Memory: 208M/683M
[INFO] ------------------------------------------------------------------------
```
https://gist.github.com/kabir/6db0a7070ea6019b1c93a1c60e01757a contains both the core and the full generated component matrix contents (There are 2 separate files in the gist). 

The original hand-coded component-matrix was removed in https://github.com/wildfly/wildfly/pull/11200https://github.com/wildfly/wildfly/pull/11200. This is a slight change from the original component matrix in that it includes the dependencies from core, while previously the user would have to include both matrices.

### CODE REVIEW

**Comment by kabir** [comment]
Retest this please

File: component-matrix-builder/pom.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    -->
+    <version>13.0.0.Alpha1-SNAPSHOT</version>
+
+    <packaging>pom</packaging>
+
+    <name>WildFly: Component matrix builder</name>
+    <description>WildFly: Dependency Component matrix BOM Builder</description>
+
+    <build>
+        <plugins>
```
Comment (rsvoboda) [inline component-matrix-builder/pom.xml]: Could be this moved to pluginManagement - https://github.com/wildfly/wildfly/blob/master/pom.xml#L6252 ?

File: component-matrix-builder/pom.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                    <execution>
+                        <id>build-bom</id>
+                        <goals>
+                            <goal>build-bom</goal>
+                        </goals>
+                        <configuration>
+                            <parent>
+                                <groupId>org.jboss</groupId>
+                                <artifactId>jboss-parent</artifactId>
+                                <version>26</version>
```
Comment (rsvoboda) [inline component-matrix-builder/pom.xml]: I don't like to have jboss-parent version defined in two places.

Can be something done about this or it is maven / plugin limitation ?

Comment (rsvoboda) [reply component-matrix-builder/pom.xml]: Maybe this could make it to the jboss-parent and define here just `<bom*>` configuration

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by rsvoboda** [comment]
Just to be clear I happy this is coming to the codebase, just trying to push it a bit further.
My comments could be taken as next step tracked in another task.

**Review by rsvoboda — APPROVED** [review]
Approving after discussion with @kabir 

My comments are tracked in https://issues.jboss.org/browse/WFLY-10370 as another task

**Pull request merged**